x-pack/plugins/data_visualizer/public/application/index_data_visualizer/components/index_data_visualizer_view/index_data_visualizer_view.tsx

```diff
@@ -381,34 +380,23 @@ export const IndexDataVisualizerView: FC<IndexDataVisualizerViewProps> = (dataVi
     };
   }, []);
 
-  // Inject custom action column for the index based visualizer
-  // Hide the column completely if no access to any of the plugins
-  const extendedColumns = useMemo(() => {
-    const actions = getActions(
-      currentIndexPattern,
-      services,
-      {
+  useEffect(
+    () => {
+      // Update data query manager if input string is updated
+      data?.query.queryString.setQuery({
+        query: searchString,
+        language: searchQueryLanguage,
+      });
+    }, // eslint-disable-next-line react-hooks/exhaustive-deps
+    [
+      data,
+      // eslint-disable-next-line react-hooks/exhaustive-deps
+      JSON.stringify({
         searchQueryLanguage,
         searchString,
-      },
-      actionFlyoutRef
-    );
-    if (!Array.isArray(actions) || actions.length < 1) return;
-
-    const actionColumn: EuiTableActionsColumnType<FieldVisConfig> = {
-      name: (
-        <FormattedMessage
-          id="xpack.dataVisualizer.index.dataGrid.actionsColumnLabel"
-          defaultMessage="Actions"
-        />
-      ),
-      actions,
-      width: '100px',
-    };
-
-    return [actionColumn];
-  }, [currentIndexPattern, services, searchQueryLanguage, searchString]);
-
+      }),
```

Comment:
Just wondering if this is necessary, searchQueryLanguage & searchString are both strings, or am I missing something? so why building an object and then converting it to another string?